test/Microsoft.AspNetCore.SystemWebAdapters.Tests/VirtualPathUtilityTests.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Web;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Tests
+{
+    public class VirtualPathUtilityTests
+    {
+        [InlineData("/", "/")]
+        [InlineData("~/", "/")]
+        [Theory]
+        public void ToAbsolute(string virtualPath, string expected)
+        {
+            Assert.Equal(expected, (string?)VirtualPathUtility.ToAbsolute(virtualPath, "/"));
+        }
+
+        [Fact]
+        public void ToAbsoluteError()
+        {
+#if NET40_OR_GREATER
+			// This does not match the documentation
+			Assert.Throws<ArgumentException>(() => VirtualPathUtility.ToAbsolute("hello", "/"));
+			Assert.Throws<ArgumentException>(() => VirtualPathUtility.ToAbsolute("../../test", "/"));
+#elif NET6_0_OR_GREATER
+            Assert.Throws<ArgumentOutOfRangeException>(() => VirtualPathUtility.ToAbsolute("hello", "/"));
+            Assert.Throws<ArgumentOutOfRangeException>(() => VirtualPathUtility.ToAbsolute("../../test", "/"));
+#endif
+        }
+
+        [InlineData("/", "~/")]
+        [InlineData("~/", "~/")]
+        [Theory]
+        public void ToAppRelative(string virtualPath, string expected)
+        {
+            Assert.Equal(expected, (string?)VirtualPathUtility.ToAppRelative(virtualPath, "/"));
+        }
+
+
+        [InlineData("~/test", "hello", "~/hello")]
+        [InlineData("~/test/world", "hello", "~/test/hello")]
+        [InlineData("~/test/world", "../hello", "~/hello")]
+        [Theory]
+        public void Combine(string basePath, string relativePath, string expected)
+        {
+            Assert.Equal(expected, (string?)VirtualPathUtility.Combine(basePath, relativePath));
+        }
+
+        [Fact]
+        public void CombineError()
+        {
+            Assert.Throws<System.Web.HttpException>(() => VirtualPathUtility.Combine("~/", "../../"));
+        }
+
+        [InlineData("", "")]        // This isn't mentioned in the docs but matches the behaviour of ASP.NET 4.x
+        [InlineData(null, null)]      // This isn't mentioned in the docs but matches the behaviour of ASP.NET 4.x
+        [InlineData("/", "/")]
+        [InlineData("/test", "/test/")]
+        [InlineData("test", "test/")]
+        [Theory]
+        public void AppendTrailingSlash(string virtualPath, string expected)
+        {
+            Assert.Equal(expected, (string?)VirtualPathUtility.AppendTrailingSlash(virtualPath));
+        }
+
+        [InlineData(null, null)]
+        [InlineData("", null)]
+        [InlineData("/", "/")]
+        [InlineData("/test/", "/test")]
+        [Theory]
+        public void RemoveTrailingSlash(string virtualPath, string expected)
+        {
+            Assert.Equal(expected, (string?)VirtualPathUtility.RemoveTrailingSlash(virtualPath));
+        }
+
+#if NET6_0_OR_GREATER
+        [InlineData("~", "/")]
```

Comment:
Why are these conditioned?

Reply:
I was running the same tests on a net48 project to ensure I was getting the same results.
The extra data is conditioned because under net48, these calls fail due to the fact that ~ is not defined unless you are running under a web application context (which the tests weren't) so they were failing on net48. I setup a webapi page that ran through the same data and showed the outputs - this is actually how I checked the exceptions and values matched.

Reply:
Ah - that makes sense part of why I haven't tended to run tests on framework. I see value in it to make sure we don't regress, so if you'd like to keep that running on .net 48 that's fine. Just add a comment as to why they're conditioned